changelog.d/12381.misc

```diff
@@ -0,0 +1 @@
+Use `poetry` to manage Synapse's dependencies.
```

Comment:
Let's start with something slightly less grand:

```suggestion
Lay groundwork for using `poetry` to manage Synapse's dependencies.
```

Hopefully we can get the rest of this landed in time for a release, in which case we can update the changelog entries.